src/Lykke.Cqrs/Middleware/EventInterceptorsProcessor.cs

```diff
@@ -24,17 +24,19 @@ internal void AddInterceptor(IEventInterceptor eventInterceptor)
             var actualHandlerInterceptor = new EventActualHandlerInterceptor(actualHandler);
             var interceptor = _eventInterceptors.FirstOrDefault() ?? actualHandlerInterceptor;
 
-            var context = new EventInterceptionContext
+            var commonContext = new EventInterceptionContext
             {
                 Event = @event,
                 HandlerObject = handlerObject,
                 CommandSender = commandSender,
-                InterceptorsProcessor = this,
-                ActualHandlerInterceptor = actualHandlerInterceptor,
             };
-            context.Next = new EventInterceptorInvokationDecorator(context, 0);
+            var interceptorContext = new EventInterceptorContext(
+                commonContext,
+                0,
+                this,
+                actualHandlerInterceptor);
 
-            return interceptor.InterceptAsync(context);
+            return interceptor.InterceptAsync(interceptorContext);
         }
 
         internal IEventInterceptor ResolveNext(int currentInterceptorIndex, IEventInterceptor actualHandlerInterceptor)
```

Comment:
Looks like actualHandlerInterceptor is not needed here. This method can just return when all interceptors in the queue are already iterated. Calling method can decide what to do in such case.

Reply:
And since this method can return null, I suggest to rename it to the TryResolveNext